lib/internal/main/repl.js

```diff
@@ -32,31 +34,42 @@ if (process.env.NODE_REPL_EXTERNAL_MODULE) {
     process.exit(1);
   }
 
-  console.log(`Welcome to Node.js ${process.version}.\n` +
-    'Type ".help" for more information.');
+  esmLoader.initializeLoader().then(() => {
```

Comment:
```suggestion
  PromisePrototypeCatch(PromisePrototypeThen(esmLoader.initializeLoader(), () => {
```

Reply:
Is any user code already executed in this stage? If not, then there's no need for the primordials.

The code is also less readable and more difficult to follow by using these primordials.

Reply:
What about the user’s loader itself?

Reply:
That would be possible. I would still argue that we should not do this. If a user actively manipulates the promises in a way that conflicts with Node.js, so be it.

I would be fine to work around this by using async / await:

```js
(async() => {
  try {
    await esmLoader.initializeLoader();
  } catch (error) {
    if (hasUncaughtExceptionCaptureCallback()) {
      process._fatalException(e);
      return;
    }
    internalBinding('errors').triggerUncaughtException(
      e,
      true /* fromPromise */
    );
  }

  ... other code
})();
```